extension/src/plots/model/index.test.ts

```diff
@@ -45,6 +46,33 @@ describe('plotsModel', () => {
     jest.clearAllMocks()
   })
 
+  it('should update outdated custom state', () => {
+    memento = buildMockMemento({
+      [PersistenceKey.PLOTS_CUSTOM_ORDER + exampleDvcRoot]: [
+        {
+          metric: 'metrics:summary.json:loss',
+          param: 'params:params.yaml:dropout'
+        }
+      ]
+    })
+    model = new PlotsModel(
+      exampleDvcRoot,
+      {
+        getFirstThreeColumnOrder: mockedGetFirstThreeColumnOrder,
+        getSelectedRevisions: mockedGetSelectedRevisions,
+        isReady: () => Promise.resolve(undefined)
+      } as unknown as Experiments,
+      memento
+    )
+    expect(model.getCustomPlotsOrder()).toStrictEqual([
```

Comment:
Couldn't really think of a simple way to test if trends state was deleted 🤔

Reply:
Instead of coupling this with the plots model data. Maybe we should add a function in `extension/persistence/util.ts` called `removedOutdatedState` with outdated keys in there. This could be called on extension load. This would be easier to test (similar to the `resetPersistedState` tests already there).

Reply:
I commented before really reading what the whole PR does, sorry. Might be a little harder to do with what we are trying to achieve at the moment.

Reply:
You can use `memento.get('key')` to see what's actually in the state.